supervisor/addons/addon.py

```diff
@@ -104,6 +114,54 @@ def __init__(self, coresys: CoreSys, slug: str):
         self.instance: DockerAddon = DockerAddon(coresys, self)
         self._state: AddonState = AddonState.UNKNOWN
 
+        @Job(
+            name=f"addon_{slug}_restart_after_problem",
+            limit=JobExecutionLimit.THROTTLE_RATE_LIMIT,
+            throttle_period=WATCHDOG_THROTTLE_PERIOD,
+            throttle_max_calls=WATCHDOG_THROTTLE_MAX_CALLS,
+            on_condition=AddonsJobError,
+        )
+        async def restart_after_problem(addon: Addon, state: ContainerState):
```

Comment:
I can't follow why this function is inline and not just called `_restart_after_problem`

Reply:
For reference that is also why I had to do this in every plugin:
https://github.com/home-assistant/supervisor/blob/473aed3f31019e59bd841e239db8a765267c3be8/supervisor/plugins/audio.py#L212-L220

Instead of just putting the method in `PluginBase` and making it a job. Then there was one job and one rate limit for all plugins.